### PR TITLE
Rittal CMC III: Add device support for PU and PU Compact

### PR DESCRIPTION
**Pull request by nightcore500**
This adds better support for Rittal CMC III PU and PU Compact. Until now, the sensors have been queried using entity-sensor-mib. Since the cmc's do not seem to support them properly, some sensor information or complete sensors are missing. This patch uses Rittal's own MIB to query all sensor information. (The cmcIIIVarTable is a complete mess) Theoretically this can be used for all Rittal CMC III variants. However, since I do not own all types (e.g. the LCP variant) I cannot guarantee this. Therefore this patch is limited to the PU and PU Compact variant.

cmc

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)
- [x] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

File: includes/discovery/sensors/rittal-cmc-iii-sensors.inc.php [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+        $sensor_data['logic'][0] = 'OK';
+        $sensor_data['logic'][1] = 'Alarm';
+    } elseif ($sensor_data['name'] == 'Access') {
+        $sensor_data['logic'][0] = 'Closed';
+        $sensor_data['logic'][1] = 'Open';
+    }
+
+    if (isset($sensor_data['logic'])) {
+        // We need separate state indexes for each device because the sensor logic can vary from device to device depending on its configuration. So we add our device ID here.
+        $sensor_data['name'] = $sensor_data['name'] . '_' . $device['device_id'];
```
Comment (nightcore500) [inline includes/discovery/sensors/rittal-cmc-iii-sensors.inc.php:112]: Looks like I will have to solve it differently. Does anyone have an idea?

**Comment by murrant** [comment]
This pull request has been mentioned on **LibreNMS Community**. There might be relevant details there:

https://community.librenms.org/t/rittal-cmciii-discovered-as-lcp/13749/3

**Comment by Jellyfrog** [comment]
Wondering if some of these sensor data can't be in yaml format? (We really prefer that as far as its possible)

**Comment by nightcore500** [comment]
> Wondering if some of these sensor data can't be in yaml format? (We really prefer that as far as its possible)

Since the cmcIIIVarTable is the only place to get the needed sensor data and it is filled with data horribly and sometimes arbitrarily, I do not know how the sensor data can be queried via yaml. I have already converted the data for hardware, serial and version of the devices to yaml. Also for the already existing rittal lcp support. But for the remaining sensors I don't see any other way out at the moment.

**Review by Jellyfrog — APPROVED** [review]

**Comment by murrant** [comment]
This pull request has been mentioned on **LibreNMS Community**. There might be relevant details there:

https://community.librenms.org/t/v1-69-release-changelog-november-2020/14124/1

**Pull request merged**